### PR TITLE
Bugs in RotatingFileOutputStreamSupplierTest and EphemeralFileSystemAbstraction

### PR DESCRIPTION
**Pull request by chrisvest**
A few other things that were revealed while working on #9229.

### CODE REVIEW

**Comment by chrisvest** [comment]
Build failed on known flaky test.

**Comment by chrisvest** [comment]
Please take a look, @MishaDemianenko

File: community/logging/src/test/java/org/neo4j/logging/RotatingFileOutputStreamSupplierTest.java [hunk 1]
```diff
@@ -314,8 +319,9 @@ public void rotationCompleted( OutputStream out )
             }
         } );
 
+        ExecutorService rotationExecutor = Executors.newSingleThreadExecutor();
```
Comment (MishaDemianenko) [inline community/logging/src/test/java/org/neo4j/logging/RotatingFileOutputStreamSupplierTest.java]: To make this even more correct what about wrapping it into `try...finally` block?

File: community/logging/src/test/java/org/neo4j/logging/RotatingFileOutputStreamSupplierTest.java [hunk 2]
```diff
@@ -415,15 +422,17 @@ public OutputStream openAsOutputStream( File fileName, boolean append ) throws I
             }
         };
 
+        ExecutorService rotationExecutor = Executors.newSingleThreadExecutor();
```
Comment (MishaDemianenko) [inline community/logging/src/test/java/org/neo4j/logging/RotatingFileOutputStreamSupplierTest.java]: same as above

**Review by MishaDemianenko — COMMENTED** [review]
One small minor comment :)

File: community/logging/src/test/java/org/neo4j/logging/RotatingFileOutputStreamSupplierTest.java (hunk 1 above)
Comment (chrisvest) [reply community/logging/src/test/java/org/neo4j/logging/RotatingFileOutputStreamSupplierTest.java]: Good idea!

**Comment by chrisvest** [comment]
@MishaDemianenko Fixed.

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**